Source/WebCore/Modules/webaudio/AudioBuffer.cpp

```diff
@@ -187,7 +189,7 @@ ExceptionOr<JSC::JSValue> AudioBuffer::getChannelData(JSDOMGlobalObject& globalO
 
     if (globalObject.worldIsNormal()) {
         if (!m_channelWrappers[channelIndex])
-            m_channelWrappers[channelIndex].setWeakly(constructJSArray());
+            m_channelWrappers[channelIndex].set(globalObject.vm(), wrapper(), constructJSArray());
```

Comment:
setWrapper() should be getting called:
```
Source/WebCore/bindings/js/JSDOMWrapperCache.h:    domObject->setWrapper(wrapper, wrapperOwner, &world);
```

It gets created when JS calls one of our APIs the first time and we need to return said wrapper to them.

Reply:
But we didn't need a wrapper before? So what changed to cause the wrapper now? Is there some metaprogramming to tell `AudioBuffer` now is a subclass of `ScriptWrappable`?

Reply:
We always had a wrapper, now the wrapper is inline though so the C++ implementation can have easy access to it.

Reply:
Ah, that's the part I was missing. Thanks.

Reply:
>Is there some metaprogramming to tell AudioBuffer now is a subclass of ScriptWrappable in the bindings?

I think we used to add `[ScriptWrappable]` to the IDL to make things work. However, it doesn't seem to be a thing anymore. I suspect it now work with some sort of meta-programming but I'll try and find out.

@basuke It would be good to validate with logging to make sure it doesn't always return null :)

Reply:
It's because of these overloads:
```
bool setInlineCachedWrapper(DOMWrapperWorld&, void*, JSDOMObject*, JSC::WeakHandleOwner*);
bool setInlineCachedWrapper(DOMWrapperWorld&, ScriptWrappable*, JSDOMObject* wrapper, JSC::WeakHandleOwner* wrapperOwner);
bool setInlineCachedWrapper(DOMWrapperWorld&, JSC::ArrayBuffer*, JSC::JSArrayBuffer* wrapper, JSC::WeakHandleOwner* wrapperOwner);
```

If you subclass `ScriptWrappable`, then it will call the second overload, which will call `setWrapper()`. Otherwise, it will still create a JS wrapper but not store it on the ScriptWrappable.

Reply:
Thanks for clarifying.